vsintegration/src/FSharp.Editor/Navigation/GoToDefinition.fs

```diff
@@ -726,4 +728,216 @@ type internal FSharpNavigation
+            DocCommentId.Type entityPath
+        | true, "F" ->
+            let parts = m.Groups[2].Value.Split('.')
+            let entityPath = parts[..(parts.Length - 2)] |> List.ofArray
+            let memberOrVal = parts[parts.Length - 1]
+            DocCommentId.Field { EntityPath = entityPath; MemberOrValName = memberOrVal; GenericParameters = 0 }
+        | _ -> DocCommentId.None
+    
+    let tryFindFieldByName (name: string) (e: FSharpEntity) =
+        e.FSharpFields
+        |> Seq.filter (
+            fun x ->
+                x.DisplayName = name
+                && not x.IsCompilerGenerated)
+        |> Seq.map (fun e -> e.DeclarationLocation)
+
+    let tryFindValByNameAndType (name: string) (symbolMemberType: SymbolMemberType) (genericParametersCount: int) (e: FSharpEntity) =
+
+        let entities = e.TryGetMembersFunctionsAndValues()
+        
+        let defaultFilter (e: FSharpMemberOrFunctionOrValue) =
+            (e.DisplayName = name || e.CompiledName = name) && e.GenericParameters.Count = genericParametersCount
+
+        let  getLocation (e: FSharpMemberOrFunctionOrValue) = e.DeclarationLocation
+
+        let filteredEntities: range seq = 
+            match symbolMemberType with
+            | SymbolMemberType.Constructor when e.IsFSharpRecord ->
+                Seq.singleton e.DeclarationLocation
+            | SymbolMemberType.Other
+            | SymbolMemberType.Method ->
+                entities
+                |> Seq.filter defaultFilter
+                |> Seq.map getLocation
+            | SymbolMemberType.Constructor ->
+                entities
+                |> Seq.filter (fun x -> defaultFilter x && x.IsConstructor)
+                |> Seq.map getLocation
+            // When navigating to property for the record, it will be in members bag for custom ones, but will be in the fields in fields.
+            | SymbolMemberType.Property when e.IsFSharpRecord ->
+                let properties = 
+                    entities
+                    |> Seq.filter (fun x -> defaultFilter x && x.IsProperty)
+                    |> Seq.map getLocation
+                let fields = tryFindFieldByName name e
+                Seq.append properties fields
+            | SymbolMemberType.Event // Events are just properties
+            | SymbolMemberType.Property ->
+                entities
+                |> Seq.filter (fun x -> defaultFilter x && x.IsProperty)
+                |> Seq.map getLocation
+        filteredEntities
+
+    interface IFSharpCrossLanguageSymbolNavigationService with
+        member _.TryGetNavigableLocationAsync(assemblyName: string, documentationCommentId: string, cancellationToken: CancellationToken) : Task<IFSharpNavigableLocation> =
+            let path = docCommentIdToPath documentationCommentId
+            backgroundTask {
+                let projects = workspace.CurrentSolution.Projects |> Seq.filter (fun p -> p.IsFSharp && p.AssemblyName = assemblyName)
```

Comment:
Is the incoming assembly name normalized or fully qualified?
Thinking about this working in both ProjectReference and PackageReference scenarios, if that is possible at all.

Reply:
Just the name of dll from the project.

Roslyn will show metadata for its package references (or sourcelink if available).

We will show metadata for package references if navigated from our project.